docs/developer-guide/working-with-rules-new.md

```diff
@@ -0,0 +1,611 @@
+# Working with Rules
+
+Each ESLint rule has two files: a source file in the `lib/rules` directory and a test file in the `tests/lib/rules` directory. Both files should be named with the rule ID (i.e., `no-eval.js` for rule ID `no-eval`) The basic source code format for a rule is:
```

Comment:
Maybe it would be good to clarify that this is only true for core rules and not for rules in plugins.